src/graphql/packet.rs

```diff
@@ -92,6 +100,44 @@ impl PacketQuery {
         )
         .await
     }
+
+    #[allow(clippy::unused_async)]
+    async fn pcap<'ctx>(&self, ctx: &Context<'ctx>, filter: PacketFilter) -> Result<Pcap> {
+        let db = ctx.data::<Database>()?;
+        let store = db.packet_store()?;
+        let mut packet_vector = Vec::new();
+        let mut pcap = Vec::new();
+        let mut key_prefix = Vec::with_capacity(filter.source.len() + TIMESTAMP_SIZE + 2);
+        key_prefix.extend_from_slice(filter.source.as_bytes());
+        key_prefix.push(0);
+        key_prefix.extend_from_slice(&filter.request_time.timestamp_nanos().to_be_bytes());
+        key_prefix.push(0);
+
+        let (start, end) = filter.time();
+
+        let iter = store.boundary_iter(
+            &lower_closed_bound_key(&key_prefix, start),
+            &upper_open_bound_key(&key_prefix, end),
+            Direction::Forward,
+        );
+        let (records, _) = collect_records(iter, 1000, &filter)?;
+
+        if !records.is_empty() {
+            for (_, packet) in records {
+                packet_vector.push(packet);
+            }
+        }
```

Comment:
Why don't you use like the below instead?

`let packet_vector = records.into_iter().map(|(_, packet)| packet).collect();`

Reply:
Done